### PR TITLE
Default observer and fake-quant for backends

### PR DESCRIPTION
**Pull request by raghuramank100**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#26627 Default observer and fake-quant for backends**
* #26625 Emulate weight and activation only quant with fake quant, numerics test
* #26624 Quantization aware training: Freeze batch norm support
* #26623 Per channel fake quant

Differential Revision: [D17518194](https://our.internmc.facebook.com/intern/diff/D17518194/)

### CODE REVIEW

File: torch/quantization/QConfig.py [hunk 1]
```diff
@@ -63,3 +65,25 @@ def __new__(cls, weight):
 
 default_qat_qconfig = QConfig(activation=default_fake_quant,
                               weight=default_weight_fake_quant)
+
+def get_default_qconfig(backend='fbgemm'):
```
Comment (dzhulgakov) [inline torch/quantization/QConfig.py]: technically - it's not even fbgemm, but rather fbgemm on per-CooperLake x86 :-\ . But I don't know how to make it nicer.

File: torch/quantization/QConfig.py [hunk 2]
```diff
@@ -63,3 +65,25 @@ def __new__(cls, weight):
...
+    if backend == 'fbgemm':
+        qconfig = QConfig(activation=HistogramObserver.with_args(reduce_range=True),
+                          weight=default_per_channel_weight_observer)
```
Comment (dzhulgakov) [inline torch/quantization/QConfig.py]: so we do per-channel only for fbgemm, not for mobile?

File: torch/quantization/QConfig.py [hunk 3]
```diff
@@ -63,3 +65,25 @@ def __new__(cls, weight):
...
+    else:
+        qconfig = QConfig(activation=HistogramObserver.with_args(reduce_range=False),
+                          weight=default_weight_observer)
+    return qconfig
+
+def get_default_qat_qconfig(backend='fbgemm'):
+    if backend == 'fbgemm':
+        qconfig = QConfig(activation=FakeQuantize.with_args(observer=default_l2_observer(reduce_range = True),
```
Comment (dzhulgakov) [inline torch/quantization/QConfig.py]: I think this will create the same instance of observer for each FakeQuant - you probably need double with_args, but it gets ugly...

File: torch/quantization/QConfig.py (hunk 2 above)
Comment (raghuramank100) [reply torch/quantization/QConfig.py]: Mobile does not support per-channel kernels currently. If and when QNNPACK supports it, we can change that.

File: torch/quantization/QConfig.py (hunk 1 above)
Comment (raghuramank100) [reply torch/quantization/QConfig.py]: I agree, perhaps just call it server and have a comment stating that you need to know about the hardware platform to ensure you are picking a good default?

Comment (raghuramank100) [reply torch/quantization/QConfig.py]: Actually with mkl-dnn not sure how to improve on this, so leaving this as is for now

**Review by dzhulgakov — APPROVED** [review]
Feel free to land this PR before the others (reorder the stack) as it's useful
